src/components/ha-spinner.ts

```diff
@@ -0,0 +1,57 @@
+import Spinner from "@shoelace-style/shoelace/dist/components/spinner/spinner.component";
+import spinnerStyles from "@shoelace-style/shoelace/dist/components/spinner/spinner.styles";
+import type { PropertyValues } from "lit";
+import { css } from "lit";
+import { customElement, property } from "lit/decorators";
+
+@customElement("ha-spinner")
+export class HaSpinner extends Spinner {
+  @property() public size?: "tiny" | "small" | "medium" | "large" | number;
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+
+    if (changedProps.has("size")) {
+      switch (this.size) {
+        case "tiny":
+          this.style.setProperty("font-size", "16px");
+          break;
+        case "small":
+          this.style.setProperty("font-size", "28px");
+          break;
+        case "medium":
+          this.style.setProperty("font-size", "48px");
+          break;
+        case "large":
+          this.style.setProperty("font-size", "68px");
+          break;
+        default:
+          if (this.size !== undefined && !isNaN(this.size)) {
+            this.style.setProperty("font-size", `${this.size}px`);
+          }
```

Comment:
Why do we add this? We can already just set the `font-size` with `css` or use the `--ha-spinner-size` variable?